src/cross_staff_offset.lua

```diff
@@ -73,34 +94,28 @@ function is_out_of_range(horiz_offset)
 end
 
 function cross_staff_offset()
-    local modifier_keys_pressed = finenv.QueryInvokedModifierKeys and
-                                        (finenv.QueryInvokedModifierKeys(finale.CMDMODKEY_ALT) or finenv.QueryInvokedModifierKeys(finale.CMDMODKEY_SHIFT))
-    if not modifier_keys_pressed or dialog_result ~= finale.EXECMODAL_OK then
-        -- use global result values so that RetainLuaState remembers them
-        dialog_result, cross_offset, non_cross_offset, layer_number = user_selects_offset()
-        if dialog_result ~= finale.EXECMODAL_OK then -- user cancelled
+    if not cross_staff_nodialog then
+        if not user_selects_offset() then -- user cancelled
             return
-        end
-        if nil ~= finenv.RetainLuaState then
-            finenv.RetainLuaState = true
+        else -- check new selections
```

Comment:
One of the things that I have been using `RetainLuaState` for is to remember where the dialog box is and move it there before opening it. (You can see an example of how to do this in the current master branch of transpose_by_step.lua.) Now that we have user settings, you could do the same with user settings. I find that the UX is greatly improved when a dialog box opens at the same location where you left it. (Once all the mixin stuff is checked in, this will be handled automatically by mixin code for modeless dialogs.)